pac4j-core/src/main/java/org/pac4j/core/http/RelativeCallbackUrlResolver.java

```diff
@@ -33,7 +33,8 @@ public String compute(final String callbackUrl, WebContext context) {
 
             sb.append(context.getScheme()).append("://").append(context.getServerName());
 
-            if (context.getServerPort() != HttpConstants.DEFAULT_PORT) {
+            if(("http".contentEquals(context.getScheme()) && context.getServerPort() != HttpConstants.DEFAULT_PORT) || 
```

Comment:
I would rewrite this test with an `equalsIgnoreCase` (instead of `contentEquals`) or maybe even better (for the master branch), add two new methods to the `ContextHelper`: `isHttp(WebContext context)` and `isHttps(WebContext context)`.